### PR TITLE
Feature/add auth rules

### PR DESCRIPTION
**Pull request by estebanmual**
In this milestone we: 
  - Installed CanCanCan in your project.
  - Add a role column to the users table.
  - A user can delete a post if it is theirs or if they have an admin role 
  - A user can delete a comment if it is theirs or if they have an admin role

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -22,6 +22,12 @@ def create
     end
   end
```
Comment (hiromataba) [inline app/controllers/posts_controller.rb:26]: - [x] you should use **CanCanCan** for Post authorization in your `Post Controller`, you may do it by adding the code below in your Post Controller
```ruby
load_and_authorize_resource
```

**Review by hiromataba — CHANGES_REQUESTED** [review]
Hi @estebanmual and @Kakalanp     :wave: :wave: 

almost-there

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Your PR has a proper title and description  :+1: 
- Your linters are passing :heavy_check_mark:  
- Profesional README file :rocket: 

## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (Kakalanp) [reply app/controllers/posts_controller.rb:26]: Thanks! Will do :grin:

**Review by AnselemOdims — APPROVED** [review]
### Hi  @estebanmual 🙋‍♀️, 
Good job in making the required changes 👏 

# STATUS APPROVED ✔
Your project is complete!  There is nothing else to say other than... it's time to merge it 🔗 🚢 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers 🥂  and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Use the tag @AnselemOdims for a quicker reply


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**